src/chompfile.rs

```diff
@@ -214,36 +213,52 @@ impl ChompTaskMaybeTemplated {
     }
     pub fn targets_vec(&self) -> Result<Vec<String>> {
         if let Some(ref target) = self.target {
-            let target_str = target.to_string();
-            validate_path(&target_str)?;
+            let target_str = resolve_path(&target.to_string());
             Ok(vec![target_str])
         } else if let Some(ref targets) = self.targets {
-            let targets = targets.clone();
-            for target in &targets {
-                validate_path(&target)?;
-            }
+            let targets = targets.clone().iter().map(|t| resolve_path(&t)).collect();
             Ok(targets)
         } else {
             Ok(vec![])
         }
     }
-    pub fn deps_vec(&self) -> Result<Vec<String>> {
+    pub fn deps_vec(&self, chompfile: &Chompfile) -> Result<Vec<String>> {
+        let names = chompfile
+            .task
+            .iter()
+            .map(|t| t.name.clone().map(|t| t).unwrap_or_default())
+            .collect::<Vec<_>>();
+
         if let Some(ref dep) = self.dep {
-            let dep_str = dep.to_string();
-            validate_path(&dep_str)?;
+            let dep_str = if names.contains(dep) || skip_special_chars(dep) {
```

Comment:
So instead of `names.contains` you should just be able to do `chompfile.task.iter().find(|&&t| t.name == dep).is_some()` kind of thing.